analyzers/tests/SonarAnalyzer.UnitTest/TestCases/DebuggerDisplayUsesExistingMembers.cs

```diff
@@ -0,0 +1,247 @@
+    [DebuggerDisplay(@"Some text
+        {SomeField}")] int WithExistingFieldVerbatimMultiLine => 1;
+
+    [DebuggerDisplay("{1 + 1}")] int WithNoMemberReferenced1 => 1;
+    [DebuggerDisplay(@"{""1"" + ""1""}")] int WithNoMemberReferenced2 => 1;
+
+    [DebuggerDisplay("{Nonexistent}")] int WithNonexistentMember1 => 1;                          // Noncompliant {{'Nonexistent' doesn't exist in this context.}}
+    //               ^^^^^^^^^^^^^^^
+    [DebuggerDisplay("1 + {Nonexistent}")] int WithNonexistentMember2 => 1;                      // Noncompliant {{'Nonexistent' doesn't exist in this context.}}
+    //               ^^^^^^^^^^^^^^^^^^^
+    [DebuggerDisplay("{Nonexistent1} bla bla {Nonexistent2}")] int WithMultipleNonexistent => 1; // Noncompliant {{'Nonexistent1' doesn't exist in this context.}}
+    //               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+    [DebuggerDisplay(@"{Nonexistent}")] int WithNonexistentMemberVerbatim => 1;                  // Noncompliant {{'Nonexistent' doesn't exist in this context.}}
+    [DebuggerDisplay(@"Some text
+        {Nonexistent}")] int WithNonexistentMemberVerbatimMultiLine1 => 1;                       // Noncompliant@-1^22#34 {{'Nonexistent' doesn't exist in this context.}}
+    [DebuggerDisplay(@"Some text {Some
+        Property}")] int WithNonexistentMemberVerbatimMultiLine2 => 1;                           // FN@-1: the new line char make the expression within braces not a valid identifier
+
+    [DebuggerDisplay(ConstantWithInvalidMember)] int WithFormatAsConstant2 => 1;                            // FN: constants are not checked
+    [DebuggerDisplay("{Non" + "Existing}")] int WithFormatAsConcatenationOfLiterals => 1;                   // FN: only simple literal supported
+    [DebuggerDisplay("{Non"
+        + "Existing}")] int WithFormatAsConcatenationOfLiteralsMultiLine => 1;                              // FN: only simple literal supported
+    [DebuggerDisplay(ConstantFragment1 + ConstantFragment2)] int WithFormatAsConcatenationOfConstants => 1; // FN: only simple literal supported
+
+    [DebuggerDisplay("{this.NonexistentProperty}")] int PropertyWithExplicitThis => 1;                      // FN: "this." not supported (valid when debugging a C# project)
+    [DebuggerDisplay("{Me.NonexistentField}")] int FieldWithExplicitThis => 1;                              // FN: "Me." not supported (valid when debugging a VB.NET project)
+    [DebuggerDisplay("{1 + NonexistentProperty}")] int ContainingInvalidMembers => 1;                       // FN: expressions not supported
+}
+
+[DebuggerDisplay("{this.ToString()}")]
+[DebuggerDisplay("{Nonexistent}")]      // Noncompliant {{'Nonexistent' doesn't exist in this context.}}
+public enum TopLevelEnum { One, Two, Three }
+
+[DebuggerDisplay("{SomeProperty}")]
+[DebuggerDisplay("{SomeField}")]
+[DebuggerDisplay("{Nonexistent}")]      // Noncompliant {{'Nonexistent' doesn't exist in this context.}}
```

Comment:
I don't think there's a need to repeat the rule message this many times. It doesn't add any new information, and in the future, if we need to change the message, then we will have to do so in more places.
I would only do it for the first issue, and after that, only for special cases (e.g. when multiple members are referenced in the same attribute).

Reply:
I have removed all the ones which didn't seem necessary.
I have left the ones where the assertion on the member name, being looked for, seemed important, such as:
- whitespaces before and after the identifier,
- non-latin chars
- multiple members
- multiline strings
- nq modifier